query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -0,0 +1,634 @@
+      List<Set<String>> results =
+          filter.getChildFilterList().stream()
+              .map(f -> getMatchingViewFilterColumns(f, viewFilterMap))
+              .collect(Collectors.toList());
+
+      Set<String> result = results.get(0);
+      for (Set<String> set : results.subList(1, results.size())) {
+        // If the operation is OR, we need to get intersection of columns from all the children.
+        // Otherwise, the operation should be AND and we can get union of all columns.
+        result =
+            filter.getOperator() == Operator.OR
+                ? Sets.intersection(result, set)
+                : Sets.union(result, set);
+      }
+      return result;
+    }
+  }
+
+  /**
+   * Method to check if the given ViewColumnFilter matches the given query filter. A match here
+   * means the view column is superset of what the query filter is looking for, need not be an exact
+   * match.
+   */
+  private boolean doesSingleViewFilterMatchLeafQueryFilter(
+      Map<String, TableColumnFilter> viewFilterMap, Filter queryFilter) {
+
+    if (queryFilter.getOperator() != Operator.IN && queryFilter.getOperator() != Operator.EQ) {
+      return false;
+    }
+
+    TableColumnFilter tableColumnFilter =
+        getLogicalColumnName(queryFilter.getLhs()).map(viewFilterMap::get).orElse(null);
+    if (tableColumnFilter == null) {
+      return false;
+    }
+
+    switch (tableColumnFilter.getOperator()) {
+      case IN:
+        return isSubSet(tableColumnFilter.getValues(), queryFilter.getRhs());
+      case EQ:
+        return isEquals(tableColumnFilter.getValues(), queryFilter.getRhs());
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported view filter operator: " + tableColumnFilter.getOperator());
+    }
+  }
+
+  /** Checks that the values from the given expression are a subset of the given set. */
+  private boolean isSubSet(Set<String> values, Expression expression) {
+    if (!expression.hasLiteral()) {
+      return false;
+    }
+
+    return values.containsAll(getExpressionValues(expression.getLiteral()));
+  }
+
+  /** Checks that the values from the given expression are a subset of the given set. */
+  private boolean isEquals(Set<String> values, Expression expression) {
+    if (!expression.hasLiteral()) {
+      return false;
+    }
+
+    return values.equals(getExpressionValues(expression.getLiteral()));
+  }
+
+  private Set<String> getExpressionValues(LiteralConstant literalConstant) {
+    Set<String> expressionValues = new HashSet<>();
+    Value value = literalConstant.getValue();
+    switch (value.getValueType()) {
+      case STRING:
+        expressionValues.add(value.getString());
+        break;
+      case STRING_ARRAY:
+        expressionValues.addAll(value.getStringArrayList());
+        break;
+      case INT:
+        expressionValues.add(String.valueOf(value.getInt()));
+        break;
+      case INT_ARRAY:
+        expressionValues.addAll(
```

Comment:
Q:are we treating `INT_ARRAY` as a set in other impl too?

Reply:
Yes, same as Pinot